proxy/ReverseProxy.cc

```diff
@@ -67,7 +67,8 @@ init_reverse_proxy()
 
   Note("%s loading ...", ts::filename::REMAP);
   if (!rewrite_table->load()) {
-    Fatal("%s failed to load", ts::filename::REMAP);
+    Warning("%s failed to load", ts::filename::REMAP);
+    return 0;
```

Comment:
I must disagree with @randall - this return will prevent unrelated configuration update handlers from being installed. See lines 74-77. The GC protection on `rewrite_table` is also lost - that might cause a dangling pointer. You might do better to add an `else`.
```
if (rewrite_table.load()) {
  Note("loaded");
} else {
  Warning("failed");
}
```
Note there should be nothing to prevent adding a valid "remap.config" file after process start. It's just another variant of reloading.

Reply:
But, the code and the configs that are being added past this if check all seem to be somehow related to remaps though?

```  REC_RegisterConfigUpdateFunc("proxy.config.url_remap.filename", url_rewrite_CB, (void *)FILE_CHANGED);
  REC_RegisterConfigUpdateFunc("proxy.config.proxy_name", url_rewrite_CB, (void *)TSNAME_CHANGED);
  REC_RegisterConfigUpdateFunc("proxy.config.reverse_proxy.enabled", url_rewrite_CB, (void *)REVERSE_CHANGED);
  REC_RegisterConfigUpdateFunc("proxy.config.http.referer_default_redirect", url_rewrite_CB, (void *)HTTP_DEFAULT_REDIRECT_CHANGED);

  // Hold at least one lease, until we reload the configuration
  rewrite_table->acquire();```

Reply:
Related, but not the same.

> It seems reasonable that we'd only enable remap reloads, if there was a valid remap during start up.

I strongly disagree with that, as it means we can't do discovery to pull in "remap.config", which is something we're looking at with regard to containerization of ATS.

Reply:
After further discussion with @SolidWallOfCode and @randall we will allow the on-the-fly updates independent of the startup remap load status.